manifests/master/backup.pp

```diff
@@ -0,0 +1,19 @@
+# @summary Configure xCAT master backups
+#
+# @example
+#   include profile_xcat::master::backup
+class profile_xcat::master::backup {
+  include ::profile_backup::client
+
+  profile_backup::client::add_job { 'profile_xcat':
+    paths            => [
+      '/etc/xcat',
+      '/install/files',
+      '/install/postscripts',
+      '/root/xcat-tools/conf',
+      '/var/backups/xcat',
+    ],
+    prehook_commands => '/root/xcat-tools/cron_scripts/backup-node_configs.sh
+/root/xcat-tools/cron_scripts/backup-xcat.sh',
```

Comment:
Are these prehook_commands meant to be two separate strings? As-is it looks like the backup-node_configs.sh and backup-xcat.sh scripts are being run as a single command

Reply:
Ahh ok I do see how they'd get setup as individual lines in your original code.

I was assuming that `prehook_commands` and `posthook_commands` over in `profile_backup::client::add_job` accepted an array, especially since the names of the variables are plural.

I see your new changes and think that looks better, but I'm wondering if the best option would be to change the `profile_backup::client::add_job` class to accept those as an array. The main benefits would be that

- Easier to use (dev doesn't have to make a separate hash and then join function)
- Easier to read (the commands are listed right where they need to be, no need to search for the separate hash)
- It will be easier to make the change now, before too many modules are using this class

Reply:
@bsper2 For now I've resolved this with:
```
    $prehook_commands_array = [
      '/root/xcat-tools/cron_scripts/backup-node_configs.sh',
      '/root/xcat-tools/cron_scripts/backup-xcat.sh',
    ]

    profile_backup::client::add_job { 'profile_xcat':
      paths            => [
        '/etc/xcat',
        '...',
      ],
      prehook_commands => join($prehook_commands_array, '; '),
    }
```
Is that better?

Reply:
Yeah I think always having it as an array is ok, keeps things simpler.

Reply:
This change has been made in https://github.com/ncsa/puppet-profile_backup/pull/4 and this repo.
This is a lot better. Thanks!